### PR TITLE
Implementation for SMB2 lock command

### PR DESCRIPTION
**Pull request by yin19941005**
Implementation for SMB2 lock command. Add validation for SMB2Lock fields value. Add error NTSTATUS for SMB2 lock command.

### CODE REVIEW

**Comment by yin19941005** [comment]
I think yes, it is possible to add an integration test. Let me check it out how to do it.

File: src/it/groovy/com/hierynomus/smbj/SMB2FileIntegrationTest.groovy [hunk 1]
```diff
@@ -185,4 +187,27 @@ class SMB2FileIntegrationTest extends Specification {
     cleanup:
     share.rm("bigfile")
   }
+
+  def "should lock and unlock the file"() {
```
Comment (hierynomus) [inline src/it/groovy/com/hierynomus/smbj/SMB2FileIntegrationTest.groovy]: Can we add some non-success tests also?

**Comment by mreiterer** [comment]
Any news on this pull request ?

**Comment by hierynomus** [comment]
@yin19941005 There is 1 open comment on this PR. I want to prevent leaking of those sequence numbers to the consumer facing API. After that it can be merged. I've just brought the code up to date with the current master.

**Comment by yin19941005** [comment]
Hi @hierynomus,

Base on my understanding, reconnecting and re-establishing a durable/resilient/persistent handle (re-establishable handle) is the duty of upper layer application. Once the re-establishable handle is implemented, I think we have to expose the lock squence index and number to the API.

But Anyway, I agree to merge this PR first and maybe have further discussion for it on re-establishable handle PR in the future.

**Comment by nddipiazza** [comment]
Hi @yin19941005  @hierynomus  this PR introduces this issue: https://github.com/hierynomus/smbj/issues/603

**Pull request merged**